### PR TITLE
doc: describe usage of log levels

### PR DESCRIPTION
**Pull request by joostjager**

### CODE REVIEW

**Comment by joostjager** [comment]
Question from offline discussion with @cfromknecht: if `error` should never happen, when do we emit `critical`?

**Comment by Roasbeef** [comment]
I totally forgot `critical` was even a thing lol. Perhaps that's reserved for errors that cause us to instantly shutdown? Like running out of disk space, or we're unable to read entropy from the CSPRNG source.

**Comment by joostjager** [comment]
Instant shutdown for critical sounds good. Then warning is for bad things that can happen (breach for example).

**Comment by cfromknecht** [comment]
> Perhaps that's reserved for errors that cause us to instantly shutdown?

Shouldn't be too hard to have critical panic similar to `log.Fatalf`

File: docs/code_contribution_guidelines.md [hunk 1]
```diff
@@ -527,6 +530,16 @@ Here's an example replacing the `lightning-onion` version checked into `lnd` wit
...
 ```
 
+<a name="LogLevels" />
+
+#### 4.10 Use of Log Levels
+
+There are six log levels available: `trace`, `debug`, `info`, `warn`, `error` and `critical`.
+
+Only use `error` for internal errors that are never expected to happen during
+normal operation. No event triggered by an external sources (rpc, chain backend,
```
Comment (bhandras) [inline docs/code_contribution_guidelines.md:540]: Is this correct though? In case of RPC error should we not log with error?

Comment (bhandras) [inline docs/code_contribution_guidelines.md:540]: nit: "by an external sources" => "by external sources"

Comment (joostjager) [reply docs/code_contribution_guidelines.md:540]: That is indeed the intention of this definition. It is what this PR proposes.

Comment (joostjager) [reply docs/code_contribution_guidelines.md]: fixed

**Comment by joostjager** [comment]
Leaving definition of `critical` out of this pr. It may be better to first implement some critical log statements to see how we want to use them.

**Pull request merged**